src/components/gridcapa-main.js

```diff
@@ -70,8 +76,22 @@ const GridCapaMain = () => {
                         )
                     );
                 });
+            fetchMinioStorageData().then((res) => {
+                setUsedDiskSpacePercentage(0);
+                var usedDiskSpace = 0;
+                var freeDiskSpace = 0;
+                res.info.servers.forEach((server) => {
+                    server.drives.forEach((drive) => {
+                        usedDiskSpace = usedDiskSpace + drive.usedspace;
+                        freeDiskSpace = freeDiskSpace + drive.availspace;
+                    });
+                });
+                setUsedDiskSpacePercentage(
+                    (usedDiskSpace / (usedDiskSpace + freeDiskSpace)) * 100
+                );
+            });
         }
-    }, [processName]);
+    }, [processName, usedDiskSpacePercentage]);
```

Comment:
remove 'usedDiskSpacePercentage' from dependencies, otherwise your useEffect will be called several times and therefore the requests will be launched several times as well.

and I don't understand why you calculate the disk usage in this useEffect, because this useEffect will be executed each time we change applications, but the disk usage is the same for all applications.